istioctl/cmd/analyze.go

```diff
@@ -191,6 +193,21 @@ istioctl experimental analyze -L
 				}
 			}
 
+			// See if we've got any content getting piped in
+			if !isatty.IsTerminal(os.Stdin.Fd()) {
```

Comment:
Should this be before AddFileKubeSource? 

Is there a standard for precedence here? My own gut feeling is that files should take precedence over stdin, but I don't feel super strongly about it and I don't know if there's a common pattern. 

Actually, what we should probably do is match the pattern used by `istioctl validate` and `istioctl kube-inject`, where we use `-` to indicate that we're reading from stdin. e.g. 
`cat my.yaml | istioctl x analyze -` 

This also means that if we mix stdin and regular files, it's immediately obvious what order they're being applied in. e.g. 
`cat my.yaml | istioctl x analyze before.yaml - after.yaml`

And like those other subcommands (e.g. in `kubeinject.go`), we should probably just use `io.Reader` as the abstraction that doesn't care if it's stdin or a file. From there we can either refactor `AddFileKubeSource` to take an `io.Reader` instead of a filename, or simply have `AddFileKubeSource` open stdin if the provided file is `-`.

Reply:
:+1: use `-`

Reply:
Yea for some reason I didn't get it in my head that the hyphen was a fake file name and not some kind of shell convention like --, this makes sense.